lib/person.rb

```diff
@@ -1,7 +1,7 @@
 require_relative 'nameable'
 
 class Person < Nameable
-  attr_reader :id
+  attr_reader :id, :rentals
   attr_accessor :name, :age
 
   def initialize(age, name = 'Unknown', parent_permission: true)
```

Comment:
- [ ] Just as I mentioned in my previous comment, a similar implementation should be present in this `Person class` as well :+1: 
- [ ]  Per the project requirements, you needed to:

> Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
> Create the has-many side of Person and Rental (a person has many rentals).

- Currently, this isn't being implemented in the `Person class`. I will suggest you include a method for adding rentals in the class. The person itself should be sent as a parameter to create the new rental thereby achieving the `has-many` association. :wink: 
- Here is a [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship%20def%20add_rental(person,%20date)) to an example of this implementation in a previous lesson :+1:

Reply:
Hello @NahnahAJ,

First, let me thank you kindly for taking the time to leave such a detailed and honest review. I really appreciate it.

My only question before starting to work on the changes is related to how the [previous lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship%20def%20add_rental(person,%20date)) teaches us how to approach the `many-to-many` relationship through the intermediate class.

I'm pretty sure I followed the example from the lesson (the relationship between `Vet` and `Animal` through the `Visit`), hence why I used `attr_reader` (`getter`) instead of `attr_accessor` (`getter` and `setter`) for the `@rentals`  😅 

And the same applies to my approach to the rest of the relationship. I also tested it to see if it was working on the `app.rb` (that I left there, with examples even) and it was working just fine.

Could you please clarify this question? I honestly just want to understand if I misunderstood the assignment in order to improve 🙏 

Once again, thank you kindly!

Reply:
Hello @mwismann Kindly note that in this case, we need to be able to read the contents of @rental and also write to it that is the main reason why we need to set both getters and setters for it. I hope you get me. :smile: This makes it a little different from the implementation of the previous lesson